ores/scoring_systems/celery_queue.py

```diff
@@ -0,0 +1,287 @@
+import logging
+import re
+from urllib.parse import urlparse
+from itertools import chain
+
+import celery
+import celery.exceptions
+import mwapi.errors
+import revscoring.errors
+from celery.signals import before_task_publish
+
+from ..errors import TimeoutError, ScoreProcessorOverloaded
+from .scoring_system import ScoringSystem
+
+logger = logging.getLogger(__name__)
+
+APPLICATIONS = []
+
+DEFAULT_CELERY_QUEUE = "celery"
+
+
+@before_task_publish.connect
+def update_sent_state(sender=None, body=None, **kwargs):
+
+    for application in APPLICATIONS:
+        task = application.tasks.get(sender)
+        backend = task.backend if task else application.backend
+
+        logger.debug("Setting state to 'SENT' for {0}".format(body['id']))
+        backend.store_result(body['id'], result=None, status="SENT")
+
+
+class CeleryQueue(ScoringSystem):
+
+    def __init__(self, *args, application, queue_maxsize=None, **kwargs):
+        super().__init__(*args, **kwargs)
```

Comment:
It looks like this is a place where we disagree.  It seems to me that python's system of flexible arguments was designed for purposes like this.  In fact, my analyzer (language-python in Atom) works just fine with these argument patterns.  Because I do not explicitly capture the arguments that are not specified, I do not have access to them in this scope and that is exactly how it should be.  

What is the real readability issue here?  It seems to me that it is trivial to read this as "There's a set of arguments and key-word-arguments that are passed directly to ScoringSystem.__init__.  See ScoringSystem.__init__ for details.